mood-analysis.rb

```diff
@@ -28,5 +35,49 @@ def analyze_mood(words)
   "05/11 Yay, yay, yay! I'm having a awfuly great day."
 ]
 
-puts analyze_mood(text[0])
-puts analyze_mood(text[1])
+
+  text.each do |entry|
+  puts "#{entry[0..4]} #{analyze_mood(entry)}"
+  end
+
+  #determine how many logged entries it takes until there have been three :-) happy days.
+  def happy_days(text)
+    good_days = 0
+    day_count = 0
+    text.each do |text|
+      if good_days < 3 #if there aren't at least 3 happy days?
+        day_count += 1
+        if analyze_mood(text) == ":-)"
+          good_days += 1
+        end
+      end
+    end
+    puts "It takes #{day_count} days for 3 happy days to occur."
+  end
```

Comment:
This method would be better if it returned the smallest unit of information as possible. In this case, just the fixnum of how many days it takes to hit 3 happy days. That way we could potentially use this method for something else instead of being locked into this text output and this specific use case.